api_app/analyzers_manager/observable_analyzers/nvd_cve.py

```diff
@@ -0,0 +1,131 @@
+import re
+
+import requests
+
+from api_app.analyzers_manager.classes import AnalyzerRunException, ObservableAnalyzer
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class NVDDetails(ObservableAnalyzer):
+    url: str = "https://services.nvd.nist.gov/rest/json/cves/2.0"
+    _nvd_api_key: str = None
+    cve_pattern = r"^CVE-\d{4}-\d{4,7}$"
+
+    @classmethod
+    def update(self) -> bool:
+        pass
+
+    def run(self):
+        headers = {}
+        if self._nvd_api_key:
+            headers.update({"apiKey": self._nvd_api_key})
+
+        try:
+            # Validate if CVE format is correct E.g CVE-2014-1234 or CVE-2022-1234567
+            if not re.match(self.cve_pattern, self.observable_name):
```

Comment:
simply put, what I mentioned is 
```suggestion
            if not settings.STAGE_CI and not re.match(self.cve_pattern, self.observable_name):
```

Reply:
Though skipping the check would reduce my work but I think I should go with adding the tests since that will be helpful in future as well.